src/Components/src/Microsoft.AspNetCore.Components/CascadingValue.cs

```diff
@@ -166,12 +168,9 @@ void ICascadingValueComponent.Unsubscribe(ComponentState subscriber)
             _subscribers.Remove(subscriber);
         }
 
-        private void NotifySubscribers()
+        private async Task NotifySubscribersAsync()
         {
-            foreach (var subscriber in _subscribers)
-            {
-                subscriber.NotifyCascadingValueChanged();
-            }
+            await Task.WhenAll(_subscribers.Select(s => s.NotifyCascadingValueChanged()));
```

Comment:
Is there a real reason to do this VS just awaiting in a loop? I'm a little anxious about how this might interact with task scheduling VS just running the notifications in order.

Reply:
If we could skip doing this at no cost, that would be worth it!

Reply:
I'll take a look at what we can do here.

Reply:
I was able to remove all the asyncness from here